packages/shadycss/src/shadow-parts.js

```diff
@@ -194,18 +319,29 @@ export function onInsertBefore(parentNode, newNode, referenceNode) {
     return;
   }
   const host = root.host;
-  const receiverScope = host.localName;
+  const hostScope = host.localName;
   const superRoot = host.getRootNode();
-  const providerScope =
+  const parentScope =
     superRoot === document ? 'document' : superRoot.host.localName;
-  for (const part of parts) {
-    part.setAttribute(
-      'shady-part',
-      formatShadyPartAttribute(
-        providerScope,
-        receiverScope,
-        part.getAttribute('part')
-      )
-    );
+  const exportMap = findExportedPartMappings(host);
```

Comment:
Have you considered how a compile/build step might be able to remove the runtime work needed in `onInsertBefore`?

Reply:
No, but let's briefly chat offline about it.

Reply:
Chatted offline, filed https://github.com/webcomponents/polyfills/issues/347.